main/src/library/Core/Io/InputStream.flix

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Rasmus Larsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Core/Io/InputStream {
+    pub opaque type InputStream = ##java.io.InputStream
+
+    ///
+    /// Represents an error when operating on an InputStream
+    ///
+    pub enum Error {
```

Comment:
Move this into `Core/Io/IOError` and rename to `IOError`. I think the cases should be a subset of:

https://doc.rust-lang.org/std/io/enum.ErrorKind.html